### PR TITLE
MGMT-4909 - add hosts' hyperthreading configuration to create/update …

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
…cluster API

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -3736,6 +3736,11 @@ definitions:
         description: List of OLM operators to be installed.
         items:
           $ref: '#/definitions/operator-create-params'
+      hypethreading_config:
```
Comment (avishayt) [inline swagger.yaml:3739]:
```suggestion
      hyperthreading:
```
(Note the missing r in hyperthreading)

File: swagger.yaml [hunk 2]
```diff
@@ -3736,6 +3736,11 @@ definitions:
...
+        type: string
+        description: indicate whether hyperthreading needs to be activate on masters, workers, both. Default - none, do not activate
```
Comment (avishayt) [inline swagger.yaml:3741]:
```suggestion
        description: Enable hyperthreading on master nodes, worker nodes, or all nodes.
```

File: swagger.yaml [hunk 3]
```diff
@@ -3736,6 +3736,11 @@ definitions:
...
+        enum: ['Masters', 'Workers', 'All', 'None']
```
Comment (avishayt) [inline swagger.yaml:3742]:
```suggestion
        enum: ['masters', 'workers', 'all']
```

File: swagger.yaml [hunk 4]
```diff
@@ -3736,6 +3736,11 @@ definitions:
...
+        default: 'None'
```
Comment (avishayt) [inline swagger.yaml:3743]:
```suggestion
```

File: swagger.yaml [hunk 5]
```diff
@@ -3876,6 +3881,11 @@ definitions:
         description: List of OLM operators to be installed.
         items:
           $ref: '#/definitions/operator-create-params'
+      hypethreading_config:
+        type: string
+        description: indicate whether hyperthreading needs to be activate on masters, workers, both.
+        enum: ['Masters', 'Workers', 'All']
+        x-nullable: true
```
Comment (avishayt) [inline swagger.yaml:3888]: Copy from above, but here we also need none, right?

File: swagger.yaml [hunk 6]
```diff
@@ -4132,6 +4142,9 @@ definitions:
         items:
           type: object
           $ref: '#/definitions/monitored-operator'
+      hypethreading_config:
+        type: string
+        description:  indicate whether hyperthreading needs to be activate on masters, workers, both. Default - none, do not activate
```
Comment (avishayt) [inline swagger.yaml:4147]: Copy from above

File: swagger.yaml (hunk 4 above)
Comment (yevgeny-shnaidman) [reply swagger.yaml]: fixed

File: swagger.yaml (hunk 5 above)
Comment (yevgeny-shnaidman) [reply swagger.yaml]: fixed

File: swagger.yaml (hunk 6 above)
Comment (yevgeny-shnaidman) [reply swagger.yaml]: fixed

File: swagger.yaml (hunk 3 above)
Comment (yevgeny-shnaidman) [reply swagger.yaml]: fixed

File: swagger.yaml (hunk 2 above)
Comment (yevgeny-shnaidman) [reply swagger.yaml]: fixed

File: swagger.yaml (hunk 1 above)
Comment (yevgeny-shnaidman) [reply swagger.yaml]: fixed

**Review by avishayt — APPROVED** [review]

**Review by avishayt — APPROVED** [review]

**Pull request merged**